### PR TITLE
Tabs component

### LINKED ISSUE #196: Tabs Component
The Tabs core component moved into Lakefront and then replace usage in Lakehouse for easier maintenance and better documentation.

### PR DESCRIPTION
**Pull request by NehalBhandariSlalom**
This PR includes Tabs core component moved into Lakefront and then replace usage in Lakehouse for easier maintenance and better documentation.
If approved, this will close #196.
Screen Shot 2021-11-29 at 6 08 01 PM

### CODE REVIEW

File: src/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    caption: string;
+}
+
+export interface TabProps {
+    /**
+     * This is to set the options which has a key and a caption.
+     */
+    options: TabDef[];
+    /**
+     * This is to set the slected tab by default.
```
Comment (jason-ransom-slalom) [inline src/Tabs/Tabs.tsx:17]: Mispelled "selected"

File: src/Tabs/__tests__/Tabs.tests.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import Immutable from 'immutable';
```
Comment (jason-ransom-slalom) [inline src/Tabs/__tests__/Tabs.tests.jsx:2]: Immutable import needs to be removed.

File: src/Tabs/__tests__/Tabs.tests.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { render, fireEvent, cleanup } from '@testing-library/react';
+
+import Tabs from '../Tabs';
+
+afterAll(cleanup);
+
+const TABS = Immutable.List([
```
Comment (jason-ransom-slalom) [inline src/Tabs/__tests__/Tabs.tests.jsx:9]: Immutable use here needs to be removed.

**Review by jason-ransom-slalom — CHANGES_REQUESTED** [review]

File: src/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
```
Comment (jason-ransom-slalom) [inline src/Tabs/Tabs.tsx:1]: Remove React import and only import what you need to be consistent with rest of application.

File: src/Tabs/Tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+     * This is to set a class on the tab component.
+     */
+    className?: string;
+}
+
+/**
+ * The Tab Component is used to render multiple tabs. The onChange event is called whenever the user clicks a different tab.
+ * The value will set the selection of the tab by default.
+ */
+const Tabs: React.FC<TabProps> = props => {
```
Comment (jason-ransom-slalom) [inline src/Tabs/Tabs.tsx:34]: import { FC } from React above and change this line to be:
```jsx
const Tabs: FC<TabProps> = props => {
```

File: src/index.ts [hunk 6]
```diff
@@ -64,6 +64,8 @@ export { default as StepFunctionGraph } from './StepFunctionGraph/Graph';
 
 export { default as StepFunctionRenderer } from './StepFunctionRenderer/StepFunctionRenderer';
 
+export { default as Tabs } from './Tabs';
```
Comment (jason-ransom-slalom) [inline src/index.ts:67]: You might also export the associate Tab typescript types (specifically the TabDef) I believe. See how other types are exported in this file if you need a reference.

**Review by jason-ransom-slalom — CHANGES_REQUESTED** [review]

**Review by adorack-tri — APPROVED** [review]

File: src/types/global.ts [hunk 7]
```diff
@@ -12,3 +12,12 @@ export interface SelectOption<T> {
...
 }
+
+
+/**
+ * This is the structure of Tabs.
+ */
+export interface TabDef {
+    key: string;
+    caption: string;
+}
```
Comment (jason-ransom-slalom) [inline src/types/global.ts:23]: nit: Add new line here.

**Review by jason-ransom-slalom — APPROVED** [review]

**Pull request merged**